presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -573,6 +573,14 @@ public static double NaN()
         return Double.NaN;
     }
 
+    @Description("constant representing NaN float")
+    @ScalarFunction("fnan")
+    @SqlType(StandardTypes.FLOAT)
+    public static long floatNaN()
```

Comment:
Instead of implementing these as `fnan()` and `finfinity()` I think we should use `cast(nan() as float)` and `cast(infinity() as float)`